borsh-schema-derive-internal/src/attribute_helpers/field.rs

```diff
@@ -0,0 +1,191 @@
+        let map_result = meta_get_by_symbol_keys(SCHEMA, meta, &SCHEMA_FIELD_PARSE_MAP)?;
+        let schema_attributes: schema::Attributes = map_result.into();
+        Ok(Variants::Schema(schema_attributes))
+    });
+
+    let f3: Box<ParseFn> = Box::new(|attr_name, meta_item_name, meta| {
+        parse_lit_into::<syn::ExprPath>(attr_name, meta_item_name, meta)
+            .map(Variants::SerializeWith)
+    });
+
+    let f4: Box<ParseFn> = Box::new(|attr_name, meta_item_name, meta| {
+        parse_lit_into::<syn::ExprPath>(attr_name, meta_item_name, meta)
+            .map(Variants::DeserializeWith)
+    });
+
+    m.insert(BOUND, f1);
+    m.insert(SCHEMA, f2);
+    m.insert(SERIALIZE_WITH, f3);
+    m.insert(DESERIALIZE_WITH, f4);
+    m
+});
+
+#[derive(Default)]
+pub(crate) struct Attributes {
+    pub bounds: Option<bounds::Bounds>,
+    pub schema: Option<schema::Attributes>,
+    pub serialize_with: Option<syn::ExprPath>,
+    pub deserialize_with: Option<syn::ExprPath>,
+}
+
+impl From<BTreeMap<Symbol, Variants>> for Attributes {
+    fn from(mut map: BTreeMap<Symbol, Variants>) -> Self {
+        let bounds = map.remove(&BOUND);
+        let schema = map.remove(&SCHEMA);
+        let serialize_with = map.remove(&SERIALIZE_WITH);
+        let deserialize_with = map.remove(&DESERIALIZE_WITH);
+        let bounds = bounds.map(|variant| match variant {
+            Variants::Bounds(bounds) => bounds,
+            _ => unreachable!("only one enum variant is expected to correspond to given map key"),
+        });
+        let schema = schema.map(|variant| match variant {
+            Variants::Schema(schema) => schema,
+            _ => unreachable!("only one enum variant is expected to correspond to given map key"),
+        });
+
+        let serialize_with = serialize_with.map(|variant| match variant {
+            Variants::SerializeWith(serialize_with) => serialize_with,
+            _ => unreachable!("only one enum variant is expected to correspond to given map key"),
+        });
+
+        let deserialize_with = deserialize_with.map(|variant| match variant {
+            Variants::DeserializeWith(deserialize_with) => deserialize_with,
+            _ => unreachable!("only one enum variant is expected to correspond to given map key"),
+        });
+        Self {
+            bounds,
+            schema,
+            serialize_with,
+            deserialize_with,
+        }
+    }
+}
+impl Attributes {
+    pub(crate) fn parse(attrs: &[Attribute], skipped: bool) -> Result<Self, syn::Error> {
+        let mut ref_attr: Option<&Attribute> = None;
+        let mut map_result = BTreeMap::new();
+        for attr in attrs {
+            if attr.path() != BORSH {
+                continue;
+            }
+            ref_attr = Some(attr);
+
+            map_result = attr_get_by_symbol_keys(BORSH, attr, &BORSH_FIELD_PARSE_MAP)?;
+        }
+
+        let result: Self = map_result.into();
```

Comment:
I feel using `iter().find()` is going to be easier to digest:

```suggestion
        let ref_attr = attrs.into_iter().find(|attr| attr.path() = BORSH);
        let result: Self = if let Some(attr) = ref_attr {
            attr_get_by_symbol_keys(BORSH, attr, &BORSH_FIELD_PARSE_MAP)?.into()
        } else {
            BTreeMap::new().into()
        };
```

P.S. Maybe you don't need to bother with the rest of the checks if `ref_attr` is None, so consider rewriting this code to short-circuit exit earlier.